src/commands/createFunctionApp/createCreateFunctionAppComponents.ts

```diff
@@ -0,0 +1,164 @@
+    }
+
+    if (!wizardContext.advancedCreation) {
+        LocationListStep.addStep(wizardContext, promptSteps);
+        wizardContext.useConsumptionPlan = true;
+        wizardContext.stackFilter = getRootFunctionsWorkerRuntime(wizardContext.language);
+        promptSteps.push(new ConfigureCommonNamesStep());
+        executeSteps.push(new ResourceGroupCreateStep());
+        executeSteps.push(new StorageAccountCreateStep(storageAccountCreateOptions));
+        executeSteps.push(new AppInsightsCreateStep());
+        if (!context.dockerfilePath) {
+            executeSteps.push(new AppServicePlanCreateStep());
+            executeSteps.push(new LogAnalyticsCreateStep());
+        }
+    } else {
+        promptSteps.push(new ResourceGroupListStep());
+        promptSteps.push(new StorageAccountListStep(
+            storageAccountCreateOptions,
+            {
+                // The account type must support blobs, queues, and tables.
+                // See: https://aka.ms/Cfqnrc
+                kind: [
+                    // Blob-only accounts don't support queues and tables
+                    StorageAccountKind.BlobStorage
+                ],
+                performance: [
+                    // Premium performance accounts don't support queues and tables
+                    StorageAccountPerformance.Premium
+                ],
+                learnMoreLink: 'https://aka.ms/Cfqnrc'
+            }
+        ));
+        promptSteps.push(new AppInsightsListStep());
+    }
+
+
+    const storageProvider = 'Microsoft.Storage';
+    LocationListStep.addProviderForFiltering(wizardContext, storageProvider, 'storageAccounts');
+    executeSteps.push(new FunctionAppCreateStep());
+
+    return {
+        wizardContext,
+        promptSteps,
+        executeSteps
+    };
+}
+
+async function getDefaultFuncVersion(context: ICreateFunctionAppContext): Promise<FuncVersion> {
+    const settingValue: string | undefined = context.workspaceFolder ? getWorkspaceSetting(funcVersionSetting, context.workspaceFolder) : getWorkspaceSettingFromAnyFolder(funcVersionSetting);
+    // Try to get VS Code setting for version (aka if they have a project open)
+    let version: FuncVersion | undefined = tryParseFuncVersion(settingValue);
+    context.telemetry.properties.runtimeSource = 'VSCodeSetting';
+
+    if (version === undefined) {
+        // Try to get the version that matches their local func cli
+        version = await tryGetLocalFuncVersion(context, undefined);
+        context.telemetry.properties.runtimeSource = 'LocalFuncCli';
+    }
+
+    if (version === undefined) {
+        version = latestGAVersion;
+        context.telemetry.properties.runtimeSource = 'Backup';
+    }
+
+    return version;
+}
+
+async function createFunctionAppWizard(wizardContext: IFunctionAppWizardContext): Promise<{ promptSteps: AzureWizardPromptStep<IAppServiceWizardContext>[], executeSteps: AzureWizardExecuteStep<IAppServiceWizardContext>[] }> {
```

Comment:
I don't disagree with you, but this is logic that was pre-existing to this PR. I'm open to changing it, but I'd rather do it in a separate PR since I wanted to keep the core logic fundamentally the same.

https://github.com/microsoft/vscode-azurefunctions/pull/4130/files#diff-3b26e2c6749863538fbe756b9eb0e37aaa43b559e370bd6daf7aaa84a871a158L233